sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricResults.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.beam.sdk.metrics;
+
+import autovalue.shaded.com.google.common.common.collect.Iterables;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * Methods for interacting with the metrics of a pipeline that has been executed. Accessed via
+ * {@link PipelineResult#metrics()}.
+ */
+@Experimental(Kind.METRICS)
+public abstract class MetricResults {
+
+  /** Retrieve the current counter value. */
+  public MetricResult<Long> getCounter(MetricNameFilter name, @Nullable String step) {
```

Comment:
For ease of use, perhaps have overloads with/without the `step` parameter, rather than forcing users to pass a `null` step.